File: strings/reverse_a_string/reverseString.java
import java.util.*;
public class reverseString
{
    public static void main(String[] args)
    {
        Scanner in = new Scanner(System.in);
        String str;

        System.out.println("Enter any string");
        str = in.nextLine();

        System.out.println("The reversed string is: " + reversestr(str));
        in.close();
    }

    public static String reversestr(String strng)
    {
        String reverse="";
        for(int i=strng.length()-1;i>=0;i--)
        {
            reverse=reverse+strng.charAt(i);
        }
        return reverse;
    }
}

File: strings/reverse_a_string/reverseString.rb
def start_program #method defination
    input = take_input_from_user
    result = reverse_string(input)
    
    puts "Reversed string is ---- #{result}" #print final result along with the new line
end

def take_input_from_user
   puts "Enter string to be reversed"
   gets.chomp #returning the input taken from user (whatever is present on last line, gets returned in ruby)
end

def reverse_string(original_string)
   reverse_string = ""
   original_string.split("").reverse.each do |char| #splitting each character of string and looping from the end
      reverse_string += char #appending each character to a new string
   end
   
   return reverse_string
end

start_program #call to execute method or start the program